modules/api-import-export/src/main/java/apim/restful/importexport/utils/APIExportUtil.java

```diff
@@ -425,66 +429,115 @@ public static void exportSequences(API api, APIIdentifier apiIdentifier, int ten
+                        writeSequenceToFile(sequenceDetails, APIConstants.API_CUSTOM_SEQUENCE_TYPE_FAULT,
+                            apiIdentifier);
                     }
                 }
-            } catch (APIManagementException e) {
-                log.error("Error while retrieving custom sequence" + e.getMessage());
-                throw new APIExportException("Error while retrieving custom sequence", e);
+            }
+        }
+    }
+
+    /**
+     * Retrieve custom sequence details from the registry
+     *
+     * @param sequenceName Name of the sequence
+     * @param type         Sequence type
+     * @param registry     Current tenant registry
+     * @return Registry resource name of the sequence and its content
+     * @throws APIExportException If an error occurs while retrieving registry elements
+     */
+    private static AbstractMap.SimpleEntry<String, OMElement> getCustomSequence(String sequenceName,
+        String type, Registry registry) throws APIExportException {
+        AbstractMap.SimpleEntry<String, OMElement> sequenceDetails;
+
+        org.wso2.carbon.registry.api.Collection seqCollection = null;
+
+        try {
+            if (APIConstants.API_CUSTOM_SEQUENCE_TYPE_IN.equals(type)) {
+                seqCollection =
+                    (org.wso2.carbon.registry.api.Collection) registry.get(APIConstants.API_CUSTOM_INSEQUENCE_LOCATION);
+            }
+            else if (APIConstants.API_CUSTOM_SEQUENCE_TYPE_OUT.equals(type)) {
+                seqCollection =
+                    (org.wso2.carbon.registry.api.Collection) registry.get(APIConstants.API_CUSTOM_OUTSEQUENCE_LOCATION);
+            }
+            else if(APIConstants.API_CUSTOM_SEQUENCE_TYPE_FAULT.equals(type)) {
+                seqCollection =
+                    (org.wso2.carbon.registry.api.Collection) registry.get(APIConstants.API_CUSTOM_FAULTSEQUENCE_LOCATION);
+            }
+
+            if (seqCollection != null) {
+                String[] childPaths = seqCollection.getChildren();
 
+                for (int i = 0; i < childPaths.length; i++) {
+                    Resource sequence = registry.get(childPaths[i]);
+                    OMElement seqElment = APIUtil.buildOMElement(sequence.getContentStream());
+                    if (sequenceName.equals(seqElment.getAttributeValue(new QName("name")))) {
+                        String sequenceFileName = sequence.getPath().substring(sequence.getPath().lastIndexOf('/'));
+                        sequenceDetails = new AbstractMap.SimpleEntry<String, OMElement>(sequenceFileName, seqElment);
+                        return sequenceDetails;
+                    }
+                }
             }
+
+        } catch (RegistryException e) {
+            log.error("Error while retrieving sequence from the registry " + e.getMessage());
+            throw new APIExportException("Error while retrieving sequence from the registry", e);
+        } catch (Exception e) {
```

Comment:
In line 494, APIUtil.buildOMElement throws a generic exception.
Therefore this clause was required to catch that exception.